source/libraries/ContractDeployer.ts

```diff
@@ -310,16 +311,20 @@ export class ContractDeployer {
 
         const reportingWindowAddress = await myUniverse.getCurrentReportingWindow.bind(constant)();
         const marketCreationFee = await marketFeeCalculator.getMarketCreationCost.bind(constant)(reportingWindowAddress);
-        const marketAddress = await marketCreation.createMarket.bind({ to: this.contracts["MarketCreation"].address, from: this.testAccounts[0].address, gas: "0x5b8d80", value: marketCreationFee, constant: true })(universe.address, endTime, numOutcomes, feePerEthInWei, denominationToken.address, numTicks, designatedReporterAddress);
+        const marketAddress = await marketCreation.createMarket.bind({ value: marketCreationFee, constant: true })(universe.address, endTime, numOutcomes, feePerEthInWei, denominationToken.address, numTicks, designatedReporterAddress);
         if (!marketAddress) {
-            throw new Error("Unable to create new categorical market.");
+            throw new Error("Unable to get address for new categorical market.");
         }
-        await marketCreation.createMarket(universe.address, endTime, numOutcomes, feePerEthInWei, denominationToken.address, numTicks, designatedReporterAddress);
-
-        return marketAddress;
+        await marketCreation.createMarket.bind({ value: marketCreationFee })(universe.address, endTime, numOutcomes, feePerEthInWei, denominationToken.address, numTicks, designatedReporterAddress);
+        const market = await parseAbiIntoMethods(this.ethjsQuery, this.signatures["Market"], { to: marketAddress, from: this.testAccounts[0].address, gas: "0x5b8d80" });
+        const marketNameHex = await padAndHexlify("Market", 64, "right");
+        if (await market.getTypeName() !== marketNameHex) {
+            throw new Error("Unable to create new categorical market");
+        }
+        return market;
     }
 
-    private async createScalarMarket(universe, endTime, feePerEthInWei, denominationToken, numTicks, designatedReporterAddress): Promise<ContractBlockchainData> {
+    private async createScalarMarket(universe, endTime, feePerEthInWei, denominationToken, numTicks, designatedReporterAddress): Promise<{ [methodName: string]: ContractMethod }> {
```

Comment:
We can delete one of these two functions and rename the other to `createMarket`.  There isn't any difference between scalar and categorical anymore (there used to be).

Reply:
Do we need to keep binary market?  Isn't it just a categorical market with a specific number of outcomes?

Reply:
I didn't see a `createBinaryMarket` function, if there is one it can be removed as well.  The `createReasonable*Market` helpers are still potentially valuable though.  They just should all call into the same `createMarket` function under the hood.

Reply:
Ok, I'll keep those but just have a single createMarket() that they call.

Reply:
Actually, I'm going to leave this as a TODO item so we can get this merged quickly.